components/feature/downloads/src/main/java/mozilla/components/feature/downloads/ui/DownloadCancelDialogFragment.kt

```diff
@@ -0,0 +1,208 @@
+package mozilla.components.feature.downloads.ui
+
+import android.annotation.SuppressLint
+import android.app.Dialog
+import android.content.Context
+import android.content.DialogInterface
+import android.graphics.Color
+import android.graphics.drawable.ColorDrawable
+import android.graphics.drawable.GradientDrawable
+import android.os.Bundle
+import android.os.Parcelable
+import android.view.Gravity
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.Window
+import android.widget.LinearLayout
+import androidx.annotation.ColorRes
+import androidx.annotation.StringRes
+import androidx.annotation.VisibleForTesting
+import androidx.appcompat.app.AppCompatDialogFragment
+import androidx.core.content.ContextCompat
+import kotlinx.parcelize.Parcelize
+import mozilla.components.feature.downloads.R
+import mozilla.components.feature.downloads.databinding.MozacDownloadCancelBinding
+
+/**
+ * The dialog warns the user that closing last private tab leads to cancellation of active private
+ * downloads.
+ */
+class DownloadCancelDialogFragment : AppCompatDialogFragment() {
+
+    var onAcceptClicked: (() -> Unit)? = null
+    var onDenyClicked: (() -> Unit)? = null
```

Comment:
nits: Are we allowing these callback to be set outside of the `newInstance`  function? :) Otherwise I think we could change the visibility to internal, what do you think?

Reply:
Yes, we actually do. Such a solution is used in AddonInstallationDialogFragment and PermissionsDialogFragment, when the app reinstatiate listeners on onStart() method, if it's possible to find a fragment via tag inside the fragment manager

Reply:
✅